### PR TITLE
8274528: Add comment to explain an HKDF optimization in SSLSecretDerivation

### PR DESCRIPTION
**Pull request by bradfordwetmore**
See comment, + a couple minor cleanups.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8274528](https://bugs.openjdk.java.net/browse/JDK-8274528): Add comment to explain an HKDF optimization in SSLSecretDerivation

### Reviewers
 * [Jamil Nimeh](https://openjdk.java.net/census#jnimeh) (@jnimeh - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5765/head:pull/5765` \
`$ git checkout pull/5765`

Update a local copy of the PR: \
`$ git checkout pull/5765` \
`$ git pull https://git.openjdk.java.net/jdk pull/5765/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5765`

View PR using the GUI difftool: \
`$ git pr show -t 5765`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5765.diff">https://git.openjdk.java.net/jdk/pull/5765.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/sun/security/ssl/SSLSecretDerivation.java [hunk 1]
```diff
@@ -141,7 +150,7 @@ public SecretKey deriveKey(String algorithm,
 
         private final byte[] label;
 
-        private SecretSchedule(String label) {
+        SecretSchedule(String label) {
```
Comment (jnimeh) [inline src/java.base/share/classes/sun/security/ssl/SSLSecretDerivation.java:153]: Why are we opening the visibility on this enum?  As far as I can tell it is only ever referenced within the SSLSecretDerivation class and could remain private if it hasn't been causing any problems up to now.

Comment (bradfordwetmore) [reply src/java.base/share/classes/sun/security/ssl/SSLSecretDerivation.java:153]: From the JLS:

Comment (bradfordwetmore) [reply src/java.base/share/classes/sun/security/ssl/SSLSecretDerivation.java:153]: From the JLS §8.8.3:  https://docs.oracle.com/javase/specs/jls/se7/html/jls-8.html#jls-8.8.3

`
If no access modifier is specified for the constructor of an enum type, the constructor is private. 
'

It's kind of like adding 'public' to interfaces.  They're not necessary.

Does that alleviate your concern?

Comment (jnimeh) [reply src/java.base/share/classes/sun/security/ssl/SSLSecretDerivation.java:153]: Yes it does.  I forgot that enums are private when there's no visibility explicitly stated.  So given that the whole thing looks good to me.

**Review by jnimeh — APPROVED** [review]
Looks good.

**Pull request closed without merging**